### DISCUSSION TITLE
Check for batch effect on data acquired with Rhapsody

### DISCUSSION THREAD

**Discussion by algebio** [satijalab/seurat#4385] [Q&A] [answered]
Hello.

I have a dataset from a Rhapsody analyzer and I'm trying to check for batch effect on the control samples. The first problem I had was to adapt the files because they come in a different format, so I splitted the original file and  I created a file with ADT data only and another with RNA only. Then I followed the tutorial for Seurat v4 "Using Seurat with multimodal data" to plot UMAPs. 

My first question is: Is this the right way to use Rhapsody data in a Seurat pipeline?
My second question is: How can I colour the UMAP by batch?

Thank you for your help.

**Comment by timoast** [comment, marked as answer]
> My first question is: Is this the right way to use Rhapsody data in a Seurat pipeline?

Seurat requires a matrix with features in the rows and cells in the columns to create an assay, if that's what you've generated than that's the correct data format.

> My second question is: How can I colour the UMAP by batch?

Use the `group.by` argument in `DimPlot()`, and see the documentation for `DimPlot()` for more information
